frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -344,8 +344,12 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
                     const incomingSnapshotByWindowId: {
                         [key: string]: eventWithTime[]
                     } = response.snapshot_data_by_window_id
+
+                    // We merge the new snapshots with the existing ones and sort by timestamp to ensure they are in order
                     Object.entries(incomingSnapshotByWindowId).forEach(([windowId, snapshots]) => {
-                        snapshotsByWindowId[windowId] = [...(snapshotsByWindowId[windowId] ?? []), ...snapshots]
+                        snapshotsByWindowId[windowId] = [...(snapshotsByWindowId[windowId] ?? []), ...snapshots].sort(
+                            (a, b) => a.timestamp - b.timestamp
```

Comment:
Is it possible to do this in the query to ClickHouse rather than post processing?

Reply:
Yep but this is the quick fix that we can get in right away. The backend fix is a little more complex due to the way the data is stored 😕